api/alarms/index.js

```diff
@@ -0,0 +1,182 @@
+let display = document.querySelector('.alarm-display');
+let log = document.querySelector('.alarm-log');
+let form = document.querySelector('.create-alarm');
+let clearButton = document.getElementById('clear-display');
+let refreshButton = document.getElementById('refresh-display');
+
+// DOM event bindings
+
+//// Alarm display buttons
+
+clearButton.addEventListener('click', () => manager.cancelAllAlarms());
+refreshButton.addEventListener('click', () => manager.refreshDisplay());
+
+//// New alarm form
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  let formData = new FormData(form);
+  let data = Object.fromEntries(formData);
+
+  // Extract form values
+  let name = data['alarm-name'];
+  let delay = Number.parseFloat(data['time-value']);
+  let delayFormat = data['time-format'];
+  let period = Number.parseFloat(data['period']);
+
+  // Prepare alarm info for creation call
+  let alarmInfo = {};
+
+  if (delayFormat === 'ms') {
+    // Specified in milliseconds, use `when` property
+    alarmInfo.when = Date.now() + delay;
+  } else if (delayFormat === 'min') {
+    // specified in minutes, use `delayInMinutes` property
+    alarmInfo.delayInMinutes = delay;
+  }
+
+  if (period) {
+    alarmInfo.periodInMinutes = period;
+  }
+
+  // Create the alarm – this uses the same signature as chrome.alarms.create
+  manager.createAlarm(name, alarmInfo);
+});
+
+class AlarmManager {
+  constructor({display, log} = {}) {
+    if (display) {
+      this.displayElement = display;
+    }
+    if (log) {
+      this.logElement = log;
+    }
+
+    this.logMessage('Manager: initializing demo');
+
+    this.displayElement.addEventListener('click', (event) => this.handleCancelAlarm(event));
+    chrome.alarms.onAlarm.addListener(alarm => this.handleAlarm(alarm));
+  }
+
+  logMessage(message) {
+    let date = new Date();
+    let pad = (val, len = 2) => val.toString().padStart(len, '0');
+    let h = pad(date.getHours());
+    let m = pad(date.getMinutes());
+    let s = pad(date.getSeconds());
+    let ms = pad(date.getMilliseconds(), 3);
+    let time = `${h}:${m}:${s}.${ms}`;
+
+    let logLine = document.createElement('div');
+    logLine.innerText = `[${time}] ${message}`;
+
+    // Log events in reverse chronological order
+    this.logElement.insertBefore(logLine, this.logElement.firstChild);
+
+    // Log events in chronological order
```

Comment:
I'd remove this commented out bit.

Reply:
Done.

I originally added it 'cause I wanted to give developers the flexibility to adjust the output based on personal preference. If we do want to provide user customization for a published version of this, it would probably be best to expose that in the options page.